### ISSUE TITLE
HTTP --> HTTPS redirects do not work

### ISSUE THREAD

**Issue by larsxschneider** [git-lfs/git-lfs#2185]
It looks like as if Git LFS does not handle redirects correctly. If I clone a Git repo via HTTP that is redirected to HTTP*S* then I observe the following error using `git lfs clone`:

```
$ git lfs clone http://github.com/larsxschneider/lfstest-individual-files.git bb
Cloning into 'bb'...
warning: redirecting to https://github.com/larsxschneider/lfstest-individual-files.git/
remote: Counting objects: 52, done.
remote: Total 52 (delta 0), reused 0 (delta 0), pack-reused 52
Unpacking objects: 100% (52/52), done.
Git LFS: (0 of 1 files) 0 B / 3.00 MB
batch response: Expected json type, got: ""
```

It also fails with `git clone`:
```
$ git clone http://github.com/larsxschneider/lfstest-individual-files.git aa
Cloning into 'aa'...
warning: redirecting to https://github.com/larsxschneider/lfstest-individual-files.git/
remote: Counting objects: 52, done.
remote: Total 52 (delta 0), reused 0 (delta 0), pack-reused 52
Unpacking objects: 100% (52/52), done.
Downloading folder/demo.file (3.00 MB)
Error downloading object: folder/demo.file (ae9d512afd1d78655599cd7c5c40d0f79fe9720ab9e02810d24fa34f093a2384)

Errors logged to /xyz/.git/lfs/objects/logs/20170428T105225.022645102.log
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: folder/demo.file: smudge filter lfs failed
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry the checkout with 'git checkout -f HEAD'
```

Log file:
```
git-lfs/2.0.2 (GitHub; darwin amd64; go 1.8)
git version 2.12.2

$ git-lfs filter-process
Error downloading object: folder/demo.file (ae9d512afd1d78655599cd7c5c40d0f79fe9720ab9e02810d24fa34f093a2384)

Smudge error: Error downloading folder/demo.file (ae9d512afd1d78655599cd7c5c40d0f79fe9720ab9e02810d24fa34f093a2384): batch response: Expected json type, got: ""
github.com/git-lfs/git-lfs/errors.newWrappedError
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/errors/types.go:166
github.com/git-lfs/git-lfs/errors.NewSmudgeError
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/errors/types.go:252
github.com/git-lfs/git-lfs/lfs.PointerSmudge
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/lfs/pointer_smudge.go:68
github.com/git-lfs/git-lfs/lfs.(*Pointer).Smudge
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/lfs/pointer.go:64
github.com/git-lfs/git-lfs/commands.smudge
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/commands/command_smudge.go:63
github.com/git-lfs/git-lfs/commands.filterCommand
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/commands/command_filter_process.go:65
github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra.(*Command).execute
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra/command.go:477
github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra.(*Command).Execute
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra/command.go:551
github.com/git-lfs/git-lfs/commands.Run
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/src/github.com/git-lfs/git-lfs/commands/run.go:68
main.main
    /private/tmp/git-lfs-20170330-80234-7hq82/git-lfs-2.0.2/git-lfs.go:35
runtime.main
    /usr/local/Cellar/go/1.8/libexec/src/runtime/proc.go:185
runtime.goexit
    /usr/local/Cellar/go/1.8/libexec/src/runtime/asm_amd64.s:2197

ENV:
LocalWorkingDir=/xyz
LocalGitDir=/xyz/.git
LocalGitStorageDir=/xyz/.git
LocalMediaDir=/xyz/.git/lfs/objects
LocalReferenceDir=
TempDir=/xyz/.git/lfs/tmp
ConcurrentTransfers=10
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic
UploadTransfers=basic
GIT_DIR=.git
```

/CC @buckelij

**Comment by ttaylorr** [comment]
@larsxschneider thanks for the report. We're planning on releasing 2.1.0 today, so I've milestoned this to be fixed in the 2.1.1 release.

**Comment by ttaylorr** [comment]
:wave: I just opened #2235 to address this.
